enumerable.rb

```diff
@@ -0,0 +1,112 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    for x in self
+      yield(x)
+    end
+    x
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    x = 0
+    for y in self
+      yield(x, y)
+      x += 1
+    end
+  end
+
+  def my_select
+    return enum_for unless block_given?
+
+    new_array = []
+    my_each do |x|
+      new_array.push(x) if yield(x)
+    end
+    new_array
+  end
+
+  def my_all?
+    return enum_for unless block_given?
+
+    count = 0
+    my_each do |x|
+      count += 1 if yield(x)
+    end
+    count == length
+  end
+
+  def my_any?
+    return enum_for unless block_given?
+
+    count = 0
+    my_each do |x|
+      count += 1 if yield(x)
+    end
+    if count.is_positive?
+      true
+    else
+      false
+    end
+  end
+
+  def my_count
+    return enum_for unless block_given?
```

Comment:
- [ ] Read the documentation to #count method again. It should return the count of all its elements if no block or param are given or count how many items equal to the param if it is given.